azure-eventhubs/src/main/java/com/microsoft/azure/eventhubs/impl/MessageReceiver.java

```diff
@@ -707,12 +706,13 @@ public ErrorContext getContext() {
                     public void onEvent() {
```

Comment:
>onEvent [](start = 32, length = 7)

what is the motivation behind this change?
This is wrong.

You are effectively scheduling a receiveLink close call and then cancelling the close-timer - without receiving response from the service.

please remember receiveLink.close() doesn't close the link immediately. It schedules a task on an internal reactor queue - for which amqp bytes will be written. We will get a callback with LinkHandler.onLinkRemoteClose, that is when the link is actually closed. Please see onError method in this file.

Reply:
So, to understand your comment: are you proposing an API change - that PartitionReceiver.close() - to be simply fire-and-forget? To understand more:
* Do you also want to reflect this in the API signature - to remove CompletableFuture as a return value for PartitionReceiver.close() method in that case? as there is nothing asynchronous in this notion?
* do you also plan to remove the closeTimer - as it is cancelled as soon as created.?

Today, without this change, if users want to achieve this, they can always fire an async task - PartitionReceiver.close() and not await on the response - so, what is the motivation of this change?

Reply:
Yes, we will need to consider updating API signature to reflect a change, but it won't be necessarily needed as returning a task doesn't have to be combined with a remote I/O operation.

Reply:
Ok, so, just to understand, is this behavior change on the Close API - planned across .net clients as well - to fix the close timeout issues due to excessive prefetch? 
or is it more holistic fixing this on the service side?